lesson_24/Auth_Django_Project/Auth/views.py

```diff
@@ -0,0 +1,58 @@
+from django.shortcuts import render
+from django.views import View
+from .models import Users
+
+
+class UsersView(View):
+    def get(self, request):
+        users = Users.objects.all()
+        private_emails = []
+        for user in users:
+            mod = str(user).split('@')
+            modded_username = mod[0][:2] + "*" * (len(mod[0]) - 2)
+            domain = mod[1]
+            modded_email = modded_username + '@' + domain
+            private_emails.append(modded_email)
+        users_count = len(users)
+        last_10_emails = private_emails[-10:]
+        return render(request, 'main.html',
+                      {'count': users_count, 'users': last_10_emails})
+
+
+class SignupView(View):
+
+    def get(self, request):
+        return render(request, 'signup.html')
+
+    def post(self, request):
+        email = request.POST['email']
+        password = request.POST['password']
+        repeat_password = request.POST['repeat_password']
```

Comment:
It would be faster if we check passwords first and if all good continue work with users